### DISCUSSION TITLE
Difference between build_dataflow_cfg and model.transform

### DISCUSSION THREAD

**Discussion by btwbtw01** [Xilinx/finn#538] [Q&A] [answered]
Hi, everyone
There are two end-to-end tutorial. 
In cybersecurity/3-build-accelerator-with-finn.ipynb example, it uses build.DataflowBuildConfig to build hardware.
![image](https://user-images.githubusercontent.com/85877169/152934909-6814b722-e835-450b-a334-4e71a0e07435.png)
![image](https://user-images.githubusercontent.com/85877169/152934970-8e87d3e8-1d64-4546-a6cd-926f7929c169.png)
In bnn-pynq/tfc_end2end_example.ipynb, it uses model.transform
![image](https://user-images.githubusercontent.com/85877169/152935277-bccc66cc-f415-41de-b88b-9c69d5ed8d85.png)
![image](https://user-images.githubusercontent.com/85877169/152935335-00d5526c-9d46-4b50-8a2b-c8baf3acd658.png)
What are the difference between these two methods?

**Comment by sakshigoel155** [comment]
Hi,
I would also like to know this.

**Comment by fpjentzsch** [comment, marked as answer]
Hi,
You can think of `build.build_dataflow_cfg()` as a wrapper around the transformations that make up the FINN compiler infrastructure. It attempts to streamline the build experience, so you don't have to manually script long sequences of transformation passes every time you to put a new model through FINN. By default, this feature performs the steps defined [here](https://github.com/Xilinx/finn/blob/dev/src/finn/builder/build_dataflow_steps.py), where each step applies transformations using `model.transform()` as you've seen it in the tfc_end2end_example notebook.

This pre-made sequence of steps doesn't work for every model, but you can still base your build flow on this feature and modify/replace/add your own steps to get going more quickly. Some FINN-examples (e.g. [VGG10](https://github.com/Xilinx/finn-examples/tree/main/build/vgg10-radioml)) use such a "semi-custom" flow for example.

**Reply by btwbtw01** [reply]
Hi, @fpjentzsch 
Thanks for your answer. So you mean the model.transform() like tidy up, streamlining, Conversion to HLS layers in tfc_end2end_example.ipynb example. You can call build.build_dataflow_cfg() function to do all of these in 3-build-accelerator-with-finn.ipynb example.
